src/_c_internal_utils.c

```diff
@@ -124,6 +124,47 @@ mpl_SetForegroundWindow(PyObject* module, PyObject *arg)
 #endif
 }
 
+static PyObject*
+mpl_SetDpiAwareness(PyObject* module)
+{
+#ifdef _WIN32
+#ifdef _DPI_AWARENESS_CONTEXTS_
+    // These functions and options were added in later Windows 10 updates, so
+    // must be loaded dynamically.
+    typedef BOOL (WINAPI *IsValidDpiAwarenessContext_t)(DPI_AWARENESS_CONTEXT);
+    typedef BOOL (WINAPI *SetProcessDpiAwarenessContext_t)(DPI_AWARENESS_CONTEXT);
+
+    HMODULE user32 = LoadLibrary("user32.dll");
+    IsValidDpiAwarenessContext_t IsValidDpiAwarenessContextPtr =
+        (IsValidDpiAwarenessContext_t)GetProcAddress(
+            user32, "IsValidDpiAwarenessContext");
```

Comment:
I intentionally made this a C module rather than a C++ module to avoid needing the casts (see how XOpenDisplay is loaded above), so you can drop the typedefs and just write
```c
BOOL WINAPI (*IsValidDpiAwarenessContextPtr)(DPI_AWARENESS_CONTEXT) =
    GetProcAddress(user32, "IsValidDpiAwarenessContext");
```
and likewise for the other.

Reply:
That still causes cast warning on Windows.

Reply:
Indeed, but (at least for me) casting the return value of GetProcAddress to (void *) fixes that.